plugins/source/aws/client/spec.go

```diff
@@ -36,6 +38,12 @@ type AwsOrg struct {
 	ChildAccountRegions         []string `json:"member_regions,omitempty"`
 }
 
+type EventBasedSync struct {
+	Account            Account    `json:"account"`
+	KinesisStreamARN   string     `json:"kinesis_stream_arn"`
+	StartTime          *time.Time `json:"start_time,omitempty"`
+	EventBasedSyncOnly bool       `json:"event_based_sync_only"`
```

Comment:
I would remove both `StartTime` and `EventBasedSyncOnly` if possible for simplicity.

If you have `event_based_sync` configured it should only do event based. If users wants a full sync they can have a separate spec without `event_based_sync`.

Same for `start_time`. Can we make so it syncs whatever that's in the stream? Otherwise users will have to track when they started and stopped the CLI. The CLI can even stop due to an unexpected error which makes it harder to track sync time and ensure the same event is not handled twice. `start_time` or any other checkpoints might be useful if there are other consumers to the stream, and we need to track what we've read.

If users want to clean a stream they can always `aws kinesis get-records --shard-iterator ...` to drop messages.

Reply:
Maybe keep `start_time` but make it so the default syncs from whatever is unprocessed in the stream

Reply:
You cannot delete messages/records out of a kinesis stream. There is no way of knowing what is processed/unprocessed